### DISCUSSION TITLE
Mobile App stopped working after changing the configurations according to the new policy of play store

### DISCUSSION THREAD

**Discussion by nidhikachhawa24** [Sunbird-Ed/Community#131] [Sunbird ED configuration discussion] [answered]
**Issue:**
Blocked to launch an app

According to the new policy of google play store,
_Your app currently targets API level 30 and must target at least API level 31 to ensure it is built on the latest APIs optimized for security and performance. Change your app's target API level to at least 31._
After changing the target API to 31 and adding android:exported="true" my build was successful but now when I'm trying to run the apk in mobile or emulator, after splash screen the screen goes blank. App is not working.

**Current Working APK:**
My current working APK has following configurations
```
Ionic:

   Ionic CLI                     : 5.4.16 (/home/ttpl-rt-58/.nvm/versions/node/v12.16.1/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 5.7.0
   @angular-devkit/build-angular : 0.901.15
   @angular-devkit/schematics    : 8.3.29
   @angular/cli                  : 9.1.15
   @ionic/angular-toolkit        : 2.3.3

Cordova:

   Cordova CLI       : 11.0.0
   Cordova Platforms : android 9.0.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.2.1, (and 47 other plugins)

Utility:

   cordova-res                          : 0.15.4
   native-run (update available: 1.7.0) : 1.6.0

System:

   Android SDK Tools : 26.1.1 (/home/ttpl-rt-58/Android/Sdk)
   NodeJS            : v12.16.1 (/home/ttpl-rt-58/.nvm/versions/node/v12.16.1/bin/node)
   npm               : 8.19.1
   OS                : Linux 5.15

```
My build.gradle
```
defaultBuildToolsVersion="29.0.3"
defaultMinSdkVersion=22
defaultTargetSdkVersion=30
defaultCompileSdkVersion=30
```
cc: @swayangjit @santoshgandham @baljeetsingh @kiran

**Comment by swayangjit** [comment]
@nidhikachhawa24 Have you tried uploading apk ?

**Reply by nidhikachhawa24** [reply]
@swayangjit I did tried, but while uploading, it is not accepting any .apk file.
Only .aab is supported in upload.

**Reply by swayangjit** [reply]
ok any error that you can see in the logs or in the console?

**Reply by nidhikachhawa24** [reply]
When the screen goes blank getting these error:
![Screenshot from 2022-09-01 13-47-47](https://user-images.githubusercontent.com/107101132/188804048-024e3fc7-2621-4abf-8f7b-996520ad158f.png)

I tried to upgrade the configurations and followed this article:
https://ionic.zendesk.com/hc/en-us/articles/7891143965975-Migrating-to-Cordova-Android-11

**Reply by swayangjit** [reply]
@nidhikachhawa24 Before upgrading it to 11 was it working ?

**Reply by nidhikachhawa24** [reply]
@swayangjit when my target API was 30 and configurations was the following, it was warking.
```
`Ionic:

   Ionic CLI                     : 5.4.16 (/home/ttpl-rt-58/.nvm/versions/node/v12.16.1/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 5.7.0
   @angular-devkit/build-angular : 0.901.15
   @angular-devkit/schematics    : 8.3.29
   @angular/cli                  : 9.1.15
   @ionic/angular-toolkit        : 2.3.3

Cordova:

   Cordova CLI       : 11.0.0
   Cordova Platforms : android 9.0.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.2.1, (and 47 other plugins)

Utility:

   cordova-res                          : 0.15.4
   native-run (update available: 1.7.0) : 1.6.0

System:

   Android SDK Tools : 26.1.1 (/home/ttpl-rt-58/Android/Sdk)
   NodeJS            : v12.16.1 (/home/ttpl-rt-58/.nvm/versions/node/v12.16.1/bin/node)
   npm               : 8.19.1
   OS                : Linux 5.15`
```

**Reply by swayangjit** [reply]
@nidhikachhawa24 ````defaultBuildToolsVersion="29.0.3"
defaultMinSdkVersion=22
defaultTargetSdkVersion=30
defaultCompileSdkVersion=30````
This is your build.gradle but it still shows the target sdk is 31.

**Reply by nidhikachhawa24** [reply]
@swayangjit 
NO
My default build.gradle was having the following configuration
`defaultBuildToolsVersion="29.0.3" defaultMinSdkVersion=22 defaultTargetSdkVersion=30 defaultCompileSdkVersion=30`
Created app bundle and tried to upload it on playstore, but 
In playstore it says that target API is required atleast 31.

When I tried to change it to 31, my build.gradle was
`defaultBuildToolsVersion="29.0.3" defaultMinSdkVersion=22 defaultTargetSdkVersion=31 defaultCompileSdkVersion=31`

**Comment by sharathkashyap** [comment, marked as answer]
@nidhikachhawa24 If in case the app needs to be targeted with new apk version. Kindly help us with BUILD logs with target version 31. We will be able to help further.

**Reply by nidhikachhawa24** [reply]
@sharathkashyap I tried to change the target version to 31. The build was successful, but not able to install the APK in Android 12.

**Comment by nidhikachhawa24** [comment]
Hello @Swayangjit Parida ***@***.***>

Can we please schedule a meeting today to debug the play store issue at my
end?

Thank You
Nidhi Kachhawa

**Reply by nidhikachhawa24** [reply]
@swayangjit 
As we've discussed in the meeting

1. I've tried to create a dummy app on play store - still unable to see the APK upload option. Tried to cross check with one of the team member also, they've told me that the APK upload option has been removed from play store.

2. Helped [@Pavithra V P](mailto:pavithra.prakash@tarento.com) to do a mobile app setup on her local. She is facing some errors in API 31 build.

**Reply by vpPavithra** [reply]
Fixed a migration to API 31 build and generated apk file successfully its working fine form ur end, and updated the same steps for @nidhikachhawa24.
@nidhikachhawa24 any updates on the migration API 31 build.

**Reply by nidhikachhawa24** [reply]
@vpPavithra 
I'm facing some local issues in terminal.
Will try to do the setup properly and then will give you update.

**Reply by luckynvdu** [reply]
@nidhikachhawa24 is your issue resolved?

**Reply by nidhikachhawa24** [reply]
@luckynvdu yes my issue is resolved

Thank you @swayangjit @vpPavithra
